### PR TITLE
Updated HLD for --json show commands and other changes

### PR DESCRIPTION
**Pull request by tsvanduyn**
Updated HLD to reflect --json show command output and additional show commands like "config snmp community replace".

### CODE REVIEW

File: doc/snmp/snmp-configdb-migration-hld.md [hunk 1]
```diff
@@ -31,7 +32,7 @@ SNMP|LOCATION
 SNMP|CONTACT
 
 admin@switch1:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
-1) "LOCATION"
+1) "Location"
```
Comment (qiluo-msft) [inline doc/snmp/snmp-configdb-migration-hld.md:35]: >Location [](http://example.com/codeflow?start=4&length=8)

Other field name are UPCASE. Do you want to align?

Comment (tsvanduyn) [reply doc/snmp/snmp-configdb-migration-hld.md:35]: I did this on purpose per what Joe asked me to do.  Also this aligns with what I've done in the commands for the other PR.

File: doc/snmp/snmp-configdb-migration-hld.md [hunk 2]
```diff
@@ -54,18 +55,18 @@ admin@switch1:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Jack"
...
 admin@switch1:~$ redis-cli -n 4 hgetall "SNMP_USER|Travis"
- 1) "SNMP_USER_ENCRYPTION_TYPE"
- 2) "AES"
- 3) "SNMP_USER_AUTH_TYPE"
- 4) "SHA"
- 5) "SNMP_USER_ENCRYPTION_PASSWORD"
- 6) "TravisEncryptPass"
+ 1) "SNMP_USER_TYPE"
+ 2) "Priv"
+ 3) "SNMP_USER_PERMISSION"
```
Comment (qiluo-msft) [inline doc/snmp/snmp-configdb-migration-hld.md:60]: The order of fields does not matter. Seem there is no change, could you keep old version?

Comment (tsvanduyn) [reply doc/snmp/snmp-configdb-migration-hld.md:60]: I'd prefer to have this match up with what we get in the redis db output that is now the finalized version.  To me that makes more sense.

File: doc/snmp/snmp-configdb-migration-hld.md [hunk 3]
```diff
@@ -237,25 +325,26 @@ Options:
 
 Commands:
   add  Add snmp user
-  del  Delete snmp user
+  del  Del snmp user
```
Comment (qiluo-msft) [inline doc/snmp/snmp-configdb-migration-hld.md:328]: >Del [](http://example.com/codeflow?start=7&length=3)

Original word is better

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please check with other reviewers.

**Review by SuvarnaMeenakshi — APPROVED** [review]

**Pull request merged**